### PR TITLE
Fix tests and add tests to travis

### PR DESCRIPTION
**Pull request by adityapb**

### CODE REVIEW

**Comment by katyhuff** [comment]
Hi @FlanFlanagan (Robert Flanagan). @adityapb (Aditya Bhosale) is a visiting undergraduate working with me at UIUC. 

@adityapb , it would be appropriate to provide a more detailed description of the purpose of this pull request so that @FlanFlanagan has some more context. It includes both stylistic changes, test improvments, and a fix to the Travis CI so that the pytest tests will also run on CI.

File: .travis.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ script:
     - cd docs && make html && cd ..
     - conda install doctr
     - doctr deploy --deploy-repo ergs/rickshaw-docs .
+    - pytest -v
```
Comment (scopatz) [inline .travis.yml]: This line should likely be above the doc creation and below the `set -e` line. We want the tests to fail first.

File: tests/test_choose_recipes.py [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-    commods = choose_commodities(niches)
+    spec = SimSpec()
+    niches = random_niches(spec, 10)
+    commods = choose_commodities(spec, niches)
     assert type(commods) is list
-    recipes = choose_recipes(commods)
+    recipes = choose_recipes(spec, commods)
     assert type(recipes) is list
-    if "mine" in niches:
+    if "mine" and "enrichment" in niches:
```
Comment (scopatz) [inline tests/test_choose_recipes.py]: I don't think that this does what you think it does.  This tests `if len("mine") > 0 and "enrichment" in niches`.  You probably want `if "mine" in niches and "enrichment" in niches`

**Comment by scopatz** [comment]
Thanks @adityapb! Just a couple of things to clean up.

**Comment by adityapb** [comment]
Made the changes. Thanks @scopatz!

**Comment by scopatz** [comment]
LGTM, thanks!

**Pull request merged**